### PR TITLE
Add `resolve_enum` to the public API.

### PR DESCRIPTION
**Pull request by OrangeChannel**
Documentation at https://vsutil-orangechannel.readthedocs.io/en/resolve-enum-public/api.html#vsutil.resolve_enum

image

Before the eventual 0.7.0 release, we need to add mention of this and setsu's three get_XXX_value functions to the changelog. If you want @kageru , I can include it in this PR?

### CODE REVIEW

**Comment by kageru** [comment]
> Before the eventual 0.7.0 release, we need to add mention of this and setsu's three get_XXX_value functions to the changelog. If you want @kageru , I can include it in this PR?

Sounds good. Just add it here in a separate commit.

File: vsutil/types.py [hunk 1]
```diff
@@ -61,11 +61,25 @@ def _readable_enums(enum: Type[Enum]) -> str:
...
+    None
+    >>> my_fn(3)
+    <ColorFamily.YUV: 3>
+    >>> my_fn(9)
+    ValueError: my_fn: family must be in <vapoursynth.UNDEFINED: 0>, <vapoursynth.GRAY: 1>, ...
+
+    :param enum:      Enumeration, i.e. ``vapoursynth.ColorFamily``.
+    :param value:     Value to check. Can be ``None``.
+    :param var_name:  User-provided parameter name that needs to be checked.
+    :param fn:        Function that should be causing the exception (used for error message only).
```
Comment (FichteFoll) [inline vsutil/types.py:80]: No `:raises:`?

File: vsutil/types.py [hunk 2]
```diff
@@ -61,11 +61,25 @@ def _readable_enums(enum: Type[Enum]) -> str:
         return ', '.join([repr(i) for i in enum])
 
 
-def _resolve_enum(enum: Type[E], value: Any, var_name: str, fn: Optional[Callable] = None) -> Union[E, None]:
+def resolve_enum(enum: Type[E], value: Any, var_name: str, fn: Optional[Callable] = None) -> Union[E, None]:
     """
-    Attempts to evaluate `value` in `enum` if value is not None, otherwise returns None.
-    Basically checks if a supplied enum value is valid and returns a readable error message
-    explaining the possible enum values if it isn't.
+    Attempts to evaluate `value` in `enum` if value is not ``None``, otherwise returns ``None``.
```
[the remark this answers is not in the document]
Comment (OrangeChannel) [reply vsutil/types.py:66]: It's more of a fix for cython's weird behavior. It should be the expected return actually if vs was a pure python module.
